src/divviProtocol/registerReferral.ts

```
@@ -22,3 +35,157 @@ export function isRegistrationTransaction(tx: TransactionRequest | SerializableT
+          )
+          return
+        }
+
+        if (registeredUsers[0].some((address) => address.toLowerCase() === walletAddress)) {
+          Logger.debug(
+            `${TAG}/createRegistrationTransactionsIfNeeded`,
+            `Referral is already registered for protocol "${protocolId}". Skipping registration transaction.`
+          )
+          store.dispatch(divviRegistrationCompleted(networkId, protocolId))
+          return
+        }
+
+        protocolsEligibleForRegistration.push(protocolId)
+      } catch (error) {
+        Logger.error(
+          `${TAG}/createRegistrationTransactionsIfNeeded`,
+          `Error reading registration state for protocol "${protocolId}". Skipping registration transaction.`,
+          error
+        )
+      }
+    })
+  )
+
+  return protocolsEligibleForRegistration.map((protocolId) => ({
+    to: REGISTRY_CONTRACT_ADDRESS,
+    data: encodeFunctionData({
+      abi: registryContractAbi,
+      functionName: 'registerReferral',
+      args: [referrerId, protocolId],
+    }),
+  }))
+}
+
+export function* sendPreparedRegistrationTransactions(
+  txs: TransactionRequest[],
+  networkId: NetworkId,
+  wallet: ViemWallet,
+  nonce: number
+) {
+  for (let i = 0; i < txs.length; i++) {
+    try {
+      const signedTx = yield* call([wallet, 'signTransaction'], {
+        ...txs[i],
+        nonce,
+      } as any)
+      const hash = yield* call([wallet, 'sendRawTransaction'], {
+        serializedTransaction: signedTx,
+      })
+
+      Logger.debug(
+        `${TAG}/sendPreparedRegistrationTransactions`,
+        'Successfully sent transaction to the network',
+        hash
+      )
+      nonce = nonce + 1
+
+      yield* spawn(monitorRegistrationTransaction, hash, networkId)
+    } catch (error) {
+      Logger.error(
+        `${TAG}/sendPreparedRegistrationTransactions`,
+        `Failed to send or parse prepared registration transaction`,
+        error
+      )
```

maybe we wanna let this bubble up? 
so we don't send the other transactions if we're not able to submit the TX for registering?
reducing the chance to have unattributed transactions